src/Sensors/IMU/IMU.h

```diff
@@ -9,26 +9,21 @@
 namespace mmfs
 {
 
-    // TODO:
-    // - Implement Global and Local orientations
-    // - Implement Global and Local accelerations
-    // - Implement Magnetometer reading fusion
-    // - Revisit naming conventions
-    // - Revisit necessary functions/variables
-
     class IMU : public Sensor
     {
     public:
         virtual ~IMU(){};
 
         //local represents the orientation off of the intial angle. Global represents the orientation with repesct to gravity
-        virtual Quaternion getOrientationLocal();
-        virtual Quaternion getOrientationGlobal();
-        virtual Vector<3> getAccelerationLocal();
+        virtual Quaternion getOrientation();
+        virtual Vector<3> getAngularVelocity();
+        virtual Vector<3> getMagField();
+        virtual Vector<3> getAcceleration();
         virtual Vector<3> getAccelerationGlobal();
-        virtual Vector<3> getMagnetometerReading();
-        virtual Vector<3> getGyroReading();
-        virtual Vector<3> getAccReading();
+        virtual double getAccelBestFilteringAtStatic() {return accel_best_filtering_at_static;};
+        virtual void setAccelBestFilteringAtStatic(double a) {accel_best_filtering_at_static = a;};
+        virtual double getMagBestFilteringAtStatic() {return mag_best_filtering_at_static;};
+        virtual void setMagBestFilteringAtStatic(double m) {mag_best_filtering_at_static = m;};
```

Comment:
What are these functions for? Are they intended to be public?

We may run into multiple definition issues with this and need to move these into the .cpp. Probably ok for now though.

Reply:
These are used to set filter constants used by the IMU and specifically the quaternionBasedComplimentaryFilter function. I will defer to you on where this should structurally live but they just need to be a getter and setter function for a double.